internal/integration/api/reset.go

```diff
@@ -233,6 +233,37 @@ func (suite *ResetSuite) TestResetWithSpecStateAndUserDisks() {
 	}, true)
 }
 
+// TestResetWithSpecEphemeralDropsPartition wipes the EPHEMERAL
+// system partition, and checks that the partition is dropped.
+func (suite *ResetSuite) TestResetWithSpecEphemeralDropsPartition() {
+	if suite.Capabilities().SecureBooted {
+		// this is because in secure boot mode, the machine config is only applied and cannot be passed as kernel args
+		suite.T().Skip("skipping as talos is explicitly trusted booted")
+	}
+
+	node := suite.RandomDiscoveredNodeInternalIP(machine.TypeWorker)
+	nodeCtx := client.WithNode(suite.ctx, node)
+
+	suite.T().Logf("resetting EPHEMERAL on node %s", node)
+
+	disks, err := suite.Client.Disks(nodeCtx)
+	suite.Require().NoError(err)
+	suite.Require().NotEmpty(disks.Messages)
+
+	suite.ResetNode(suite.ctx, node, &machineapi.ResetRequest{
+		Reboot:   true,
+		Graceful: true,
+		SystemPartitionsToWipe: []*machineapi.ResetPartitionSpec{
+			{
+				Label: constants.EphemeralPartitionLabel,
+				Wipe:  true,
+			},
+		},
+	}, true)
+
+	rtestutils.AssertNoResource[*block.DiscoveredVolume](nodeCtx, suite.T(), suite.Client.COSI, constants.EphemeralPartitionLabel)
```

Comment:
I _think_ this won't work as the ResetNode above will only return after reboot 

I don't think we can assert on this condition (and I think we already have a test which wipes by system labels)

Reply:
I'll drop the test for now – I think that should be fine, we have other tests to at least ensure that the change doesn't break anything.